packages/smooth_app/lib/helpers/analytics_helper.dart

```diff
@@ -93,29 +109,16 @@ class AnalyticsHelper {
   static String? get uuid =>
       kDebugMode ? 'smoothie-debug--' : OpenFoodAPIConfiguration.uuid;
 
-  // TODO(m123): Matomo removes leading 0 from the barcode
-  static void trackScannedProduct({required String barcode}) =>
-      MatomoTracker.instance.trackEvent(
-        eventName: _scanAction,
-        eventCategory: 'Scan',
-        action: 'Scanned',
-        eventValue: _formatBarcode(barcode),
-      );
-
-  static void trackProductPageOpen({required Product product}) =>
-      MatomoTracker.instance.trackEvent(
-        eventName: _productPageAction,
-        action: 'opened',
-        eventCategory: 'Page',
-        eventValue: _formatBarcode(product.barcode!),
-      );
-
-  static void trackPersonalizedRanking(int count) =>
+  static void trackEvent(
+    AnalyticsMessage msg, {
+    int? eventValue,
+    String? barcode,
+  }) =>
       MatomoTracker.instance.trackEvent(
-        eventName: _personalizedRankingAction,
-        action: 'opened',
-        eventCategory: 'Page',
-        eventValue: count,
+        eventName: msg.name,
+        eventCategory: msg.category.tag,
+        action: msg.name,
```

Comment:
So the action has no real added-value for us, but Matomo needs it, right?

Reply:
Correct matomo needs it 👍🏼